spras/containers.py

```diff
@@ -109,24 +110,31 @@ def prepare_dsub_cmd(flags: dict):
 
     # iteratively add flags to the command
     for flag in ordered_flags.keys():
-        if isinstance(ordered_flags.get(flag), list):
-            for f in ordered_flags.get(flag):
+        value = ordered_flags.get(flag)
+        if isinstance(value, list):
+            for f in value:
                 dsub_command = dsub_command + " --" + flag + " " + f
         else:
-            dsub_command = dsub_command + " --" + flag + " " + ordered_flags.get(flag)
+            dsub_command = dsub_command + " --" + flag + " " + value
 
     # Wait for dsub job to complete
     dsub_command = dsub_command + " --wait"
     print(f"dsub command: {dsub_command}")
     return dsub_command
 
+def env_to_items(environment: dict[str, str]) -> Iterator[str]:
+    """
+    Turns an environment variable dictionary to KEY=VALUE pairs.
+    """
+    # TODO: we should also handle special escaping here.
+    return (f"{key}={value}" for key, value in environment.items())
 
 # TODO consider a better default environment variable
 # TODO environment currently a single string (e.g. 'TMPDIR=/OmicsIntegrator1'), should it be a list?
 # run_container_singularity assumes a single string
 # Follow docker-py's naming conventions (https://docker-py.readthedocs.io/en/stable/containers.html)
 # Technically the argument is an image, not a container, but we use container here.
-def run_container(framework: str, container_suffix: str, command: List[str], volumes: List[Tuple[PurePath, PurePath]], working_dir: str, environment: str = 'SPRAS=True'):
+def run_container(framework: str, container_suffix: str, command: List[str], volumes: List[Tuple[PurePath, PurePath]], working_dir: str, environment: Optional[dict[str, str]] = None):
```

Comment:
[Note: various environment sets were put to still have defaults declared in-code to satisfy ruff, as mutable defaults can be mutated. There was also no `frozendict` to help here.]